app/questionnaire/questionnaire_schema.py

```diff
@@ -725,6 +753,23 @@ def get_first_answer_id_for_block(self, block_id: str) -> str:
         answer_ids = self.get_answer_ids_for_block(block_id)
         return answer_ids[0]
 
+    def get_answer_format_for_calculated_summary(
+        self, calculated_summary_block_id: str
+    ) -> dict:
+        """
+        Given a calculated summary block id, find the format of the total by using the first answer
+        """
+        # Type ignore: the block will exist for any valid calculated summary id
+        calculated_summary_block: ImmutableDict = self.get_block(calculated_summary_block_id)  # type: ignore
+        first_answer_id = get_calculated_summary_answer_ids(calculated_summary_block)[0]
+        first_answer = self.get_answers_by_answer_id(first_answer_id)[0]
+        return {
```

Comment:
Is it valid for an integer to be summed with a decimal for a grand calculation? Or will the calculated summaries always be same type and thus this is safe every time?

Reply:
the calculated summaries will always be the same type, currency unit, etc, and the GCS re-uses the validation for answer types, so it would never sum up a unit and a currency, but it does support an integer with decimal places, just like calculated summaries does